compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -317,17 +318,30 @@ class ResolverImpl(
             is PsiMethod -> {
                 // TODO: get rid of hardcoded check if possible.
                 if (psi.name.startsWith("set") || psi.name.startsWith("get")) {
-                    moduleClassResolver.resolveClass(JavaMethodImpl(psi).containingClass)
-                            ?.unsubstitutedMemberScope!!.getDescriptorsFiltered().singleOrNull{
-                                (it as? PropertyDescriptor)?.getter?.findPsi() == psi || (it as? PropertyDescriptor)?.setter?.findPsi() == psi
-                            }
+                    moduleClassResolver
+                        .resolveClass(JavaMethodImpl(psi).containingClass)
+                        ?.findEnclosedDescriptor(
+                            kindFilter = DescriptorKindFilter.CALLABLES
+                        ) {
+                            (it as? PropertyDescriptor)?.getter?.findPsi() == psi || (it as? PropertyDescriptor)?.setter?.findPsi() == psi
+                        }
                 } else {
-                    moduleClassResolver.resolveClass(JavaMethodImpl(psi).containingClass)
-                            ?.unsubstitutedMemberScope!!.getDescriptorsFiltered().singleOrNull { it.findPsi() == psi }
+                    moduleClassResolver
+                        .resolveClass(JavaMethodImpl(psi).containingClass)
+                        ?.findEnclosedDescriptor(
+                            kindFilter = DescriptorKindFilter.FUNCTIONS,
+                            psi = psi
```

Comment:
It looks no difference from just pass the filter lambda at this call site.

Reply:
yea you are right, i first had the psi version then added the filter version hence this unnecessary duplication.
removed the psi version to always require the lambda.